otter/integration/tests/test_convergence.py

```diff
@@ -825,6 +824,35 @@ def test_group_config_update_triggers_convergence(self):
                 maxEntities=max_servers + 2)
         )
 
+    @tag("CATC-009")
+    def test_convergence_fixes_errored_building_servers(self):
+        """
+        CATC-009
+
+        If a server transitions into ERROR status from BUILD status,
+        convergence will clean it up and create a new server to replace it.
+
+        Checks nova to make sure that convergence has not overprovisioned.
+        """
+        group = self.helper.create_group(
+            image_ref=image_ref, flavor_ref=flavor_ref,
+            min_entities=2, max_entities=10,
+            server_name_prefix="build-to-error"
+        )
+        d = MimicNova(pool=self.helper.pool).sequenced_behaviors(
+            self.rcs,
+            criteria=[{"server_name": "build-to-error.*"}],
+            behaviors=[
+                {"name": "error", "parameters": {}},
+                {"name": "default"}
+            ])
+        d.addCallback(
+            lambda _: self.helper.start_group_and_wait(group, self.rcs))
+        d.addCallback(wait_for_servers, pool=self.helper.pool, group=group,
+                      matcher=HasLength(2), timeout=600)
+        d.addCallback(lambda _: group)
```

Comment:
@kivattik @radix This is one of the tests that also need to be run with a CLB - https://github.com/rackerlabs/otter/blob/master/otter/integration/tests/test_convergence.py#L962

I return the group so that at the end, the active servers can be verified to be on the CLB.  We could also just set the group to be on the test case instead, and change that requirement that all tests that are copied over should have set `self.group`.  This just seemed slightly more explicit, although I'm not particularly happy about the requirement that THESE tests return a group.  I'm not sure whether setting an attribute on the test case or returning a group is better.

Other suggestions would also be welcome.